frontend/src/components/SharedProjectPage/elements.js

```diff
@@ -0,0 +1,292 @@
+import React, { useEffect, useState, useContext } from 'react'
+import { arrayOf, bool, objectOf, string, object, number } from 'prop-types'
+import { useRouter } from 'next/router'
+import { uniqBy } from 'lodash'
+import { Button, Form, Input, Loader, Segment, TextArea } from 'semantic-ui-react'
+
+import useForm from '@hooks/useForm'
+import ProjectUpdateFormModel from '@models/ProjectUpdateForm'
+import { useDefaultBranchFiles } from '@hooks/Gitea'
+import { commitFiles } from '@utils/giteaInternalApi'
+import { canCommit, repoExists, updateRepo } from '@utils/giteaApi'
+import { AuthContext } from '@contexts/AuthContext'
+import BoardShowcase from '@components/Board/BoardShowcase'
+import BoardExtraMenus from '@components/Board/BoardExtrasMenu'
+import OrderPCBs from '@components/Board/OrderPCBs'
+import BuyParts from '@components/Board/BuyParts/index'
+import InfoBar from '@components/Board/InfoBar'
+import Readme from '@components/Board/Readme'
+import UploadModal from '@components/UploadModal'
+
+const PageElements = ({
+  assetsPath,
+  repoFiles,
+  hasUploadPermission,
+  hasIBOM,
+  kitspaceYAML,
+  boardBomInfo,
+  zipUrl,
+  renderedReadme,
+  boardSpecs,
+  isNew,
+  previewOnly,
+  owner,
+  name,
+  description,
+  projectFullname,
+  url,
+  boardAssetsExist,
+  readmeExists,
+  kitspaceYAMLExists,
+}) => {
+  // The files in the Gitea repo associated with this project and the newly loaded files
+  const {
+    files: remoteFiles,
+    isLoading,
+    mutate,
+  } = useDefaultBranchFiles(projectFullname, { initialData: repoFiles })
+  // Details(name, path, last_modified, etc...) for files dropped in the update page
```

Comment:
nit: It's common to put a space before opening brackets when writing plain English. The way it looks made me think it was referring to some `Details` function.

Reply:
Resolved in 1c573d2.